crates/clarirs_core/src/algorithms/simplify.rs

```diff
@@ -108,26 +108,124 @@ pub fn simplify<'c>(ast: &AstRef<'c>) -> Result<AstRef<'c>, ClarirsError> {
                 _ => ctx.udiv(&lhs, &rhs)?,
             }
         }
-        AstOp::SDiv(_, _) => todo!(),
+        AstOp::SDiv(lhs, rhs) => {
+            simplify!(lhs, rhs);
+            match (lhs.op(), rhs.op()) {
+                (AstOp::BVV(lhs), AstOp::BVV(rhs)) => {
+                    ctx.sdiv(&ctx.bvv(lhs.clone())?, &ctx.bvv(rhs.clone())?)?
+                }
+                _ => ctx.sdiv(&lhs, &rhs)?,
+            }
+        }
         AstOp::URem(lhs, rhs) => {
             simplify!(lhs, rhs);
             match (lhs.op(), rhs.op()) {
                 (AstOp::BVV(lhs), AstOp::BVV(rhs)) => ctx.bvv(lhs.clone() % rhs.clone())?,
                 _ => ctx.urem(&lhs, &rhs)?,
             }
         }
-        AstOp::SRem(_, _) => todo!(),
-        AstOp::Pow(_, _) => todo!(),
-        AstOp::LShL(_, _) => todo!(),
-        AstOp::LShR(_, _) => todo!(),
-        AstOp::AShL(_, _) => todo!(),
-        AstOp::AShR(_, _) => todo!(),
-        AstOp::RotateLeft(_, _) => todo!(),
-        AstOp::RotateRight(_, _) => todo!(),
-        AstOp::ZeroExt(_, _) => todo!(),
-        AstOp::SignExt(_, _) => todo!(),
-        AstOp::Extract(_, _, _) => todo!(),
-        AstOp::Concat(_, _) => todo!(),
+        AstOp::SRem(lhs, rhs) => {
+            simplify!(lhs, rhs);
+            match (lhs.op(), rhs.op()) {
+                (AstOp::BVV(lhs), AstOp::BVV(rhs)) => {
+                    ctx.srem(&ctx.bvv(lhs.clone())?, &ctx.bvv(rhs.clone())?)?
```

Comment:
This is the same as the default case right? Instead if both of these are BVV, we want to calculate the actual signed remainder value using the type from clarirs_num and return a BVV instead of another SRem. This goes for all of the cases.